packages/yew/src/virtual_dom/vlist.rs

```diff
@@ -14,7 +15,7 @@ enum FullyKeyedState {
 #[derive(Clone, Debug)]
 pub struct VList {
     /// The list of child [VNode]s
-    pub(crate) children: Vec<VNode>,
+    pub(crate) children: Option<Rc<Vec<VNode>>>,
```

Comment:
Maybe that's not relevant but I was thinking of using IArray would help here?

I see 2 possible values but I have not investigated in details:

1. IArray uses `Rc<[T]>` instead of `Rc<Vec<T>>`. I guess that would save an allocation somewhere.
2. IArray has a variant `Static(&'static [T])`, it can maybe used in place of `None` here

Reply:
(The only issue is the hard requirement of VNode to implement ImplicitClone but I think we can allow that because in the end this would be desirable, see discussion #3022)

Reply:
I think `VList` implements `DerefMut<Target = Vec<Node>>` which allows pushing new items without cloning the entire array when the reference count is 1. Which we may not be able to use IArray if it uses slices.

Reply:
Now that you mention it, I think I had issues when converting the code because of that haha